src/main/java/de/tum/in/www1/artemis/service/util/structureoraclegenerator/JavaClassDiff.java

```diff
@@ -333,7 +334,7 @@ else if (solutionList == null || templateList == null) {
         if (solutionList.size() != templateList.size()) {
             return false;
         }
-        return solutionList.containsAll(templateList);
+        return new HashSet<>(solutionList).containsAll(templateList);
```

Comment:
Why is this necessary?

Reply:
IntelliJ marks `solutionList.containsAll(templateList)` yellow saying "Call to 'list.containsAll(collection)' may have poor performance" and suggests using `new HashSet<>(solutionList).containsAll(templateList);` instead.